### ISSUE TITLE
Promises not resolving in Safari 10.0.1

### ISSUE THREAD

**Issue by berbaquero** [not-an-aardvark/snoowrap#80]
Hello! Thanks for creating and maintaining great library! 🚀

Seems to be working fine in all modern browsers, except for **Safari 10.0.1**, unfortunately.

I generate the snoowrap instance "manually" by `POST`-ing against the reddit auth endpoint myself and then passing the refresh token to snoowrap.

```js
const r = new snoowrap({
  userAgent: "reeddit",
  clientId: clientID,
  clientSecret: "",
  refreshToken: refreshToken, // got "manually" - not using snoowrap's inner function
});
```

So far, this works fine in all browsers.

But then, simply trying this:

```js
Promise.all([
    r.getSubscriptions(),
    r.getMe().getMultireddits(),
  ]).then(data => {
    console.log('Got all data!', data); // This never fires
  });
```
doesn't work on Safari (both iOS and macOS versions), but it does on Firefox and Chrome, for instance.

Could this be a bug or maybe I'm missing something? Thanks!

**Comment by not-an-aardvark** [comment]
I'm not sure exactly what's causing that, but if it works in Firefox and Chrome and not Safari, my guess is that it's a bug in Safari.

Does it still fail if you replace it with this?

```js
r.config({ proxies: false });

Promise.all([
    r.getSubscriptions(),
    r.getMe().then(me => me.getMultireddits()),
  ]).then(data => {
    console.log('Got all data!', data); // This never fires
  });
```

**Comment by berbaquero** [comment]
Ah, you mean that maybe it's a Proxy-related issue?

It seems like indeed that's the case — running your suggested code above makes it work fine in Safari.

**Comment by not-an-aardvark** [comment]
Yes, that would be my guess. If you have time to debug, it might also be useful to browser developers if you can narrow down the issue and report a bug in Safari.
